admin_scripts/Dell/firmware_version_check.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# SVCPLAN-5154: Dell firmware version check
+# Takes a node range as input, and uses racadm to list out hardware type, bios version and iDRAC version.
+
+racadmpath=/root/xcat-tools/admin_scripts/Dell/racadm.sh
+
+# If user does not provide a node range, prompt them for one here. Invalid entries will cause the script to run against all nodes.
+if [ -z "$1" ]; then
+  read -p "Please enter a node range: " RANGE
+else
+  RANGE=$1
+fi
+
+# Assign the node range into an array to loop through
+readarray -t ARRAY < <(nodels $RANGE)
+
+# Print table headers with formatting
+printf "Running scan on noderange: $RANGE \n"
+printf "%15s %10s %-15s %-19s %-20s \n" "Nodename" ""  "Bios" "iDRAC" "Model"
+printf "__________________________________________________________________________\n"
+
+for i in "${ARRAY[@]}"
+do
+  # Run racadm <node> getversion and pull bios & iDRAC versions. Store these into a temp array for formatting.
+  readarray -t TEMP < <(/bin/bash $racadmpath $i getversion  | awk '{print $4}' | grep . | head -n 2)
+
+  printf "%15s %10s %-15s %-15s" "$i" "" "${TEMP[0]}" "${TEMP[1]}"
+  printf "%-5s"
+
+  # Pull hardware model number through dmidecode, and filter out "Poweredge" out of model name.
+  ssh $i dmidecode -s system-product-name | awk '{ print $2 }'
```

Comment:
You might want to add a `timeout` to this so that unresponsive nodes don't sit there forever.

```suggestion
  timeout 15 ssh $i dmidecode -s system-product-name | awk '{ print $2 }'
```